src/bucket/BucketTests.cpp

```diff
@@ -81,6 +81,57 @@ countEntries(std::shared_ptr<Bucket> bucket)
     auto pair = bucket->countLiveAndDeadEntries();
     return pair.first + pair.second;
 }
+
+void
+checkBucketCount(BucketList& bl, uint32_t ledgerSeq, uint32_t level, bool isCurr)
```

Comment:
I don't think this function actually counts anything: it's checking if the bucket's lower bound matches its ledger entries.
Also, can't we also check the upper bound at the same time?